grapple/types/pages.py

```diff
@@ -117,28 +117,41 @@ class Meta:
         interfaces = (PageInterface,)
 
 
-def get_specific_page(id, slug, token, content_type=None):
+def get_specific_page(id=None, slug=None, url=None, token=None, content_type=None):
     """
-    Get a spcecific page, also get preview if token is passed
+    Get a specific page, also get preview if token is passed
     """
     page = None
     try:
+        # Query page based using id, slug, url
+        pages = WagtailPage.objects.live().public()
         if id:
-            page = WagtailPage.objects.live().public().specific().get(pk=id)
+            page = pages.get(id=id)
         elif slug:
-            page = WagtailPage.objects.live().public().specific().get(slug=slug)
+            page = pages.get(slug=slug)
+        elif url:
+            pages = pages.filter(url_path__contains=url.lstrip("/"))
```

Comment:
so, if you have

two pages: `/example-blog/` and `/an/example-blog/`
this will return them both, and given https://github.com/wagtail/wagtail/blob/89313120257cfafe8a5a3d9132da9377b37a7de1/wagtail/core/models.py#L203, you will get `/an/example-blog/`. Is that the expected outcome? We probably want to document this

Reply:
the solution, of course is to account for the full url with the homepage path prefix.
Will work on this next week